public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -136,15 +137,29 @@ export class PrometheusDatasource extends DataSourceApi<PromQuery, PromOptions>
   }
 
   // Use this for tab completion features, wont publish response to other components
-  metadataRequest<T = any>(url: string) {
+  async metadataRequest<T = any>(url: string) {
     const data: any = {};
     for (const [key, value] of this.customQueryParameters) {
       if (data[key] == null) {
         data[key] = value;
       }
     }
 
-    return this._request<T>(url, data, { method: 'GET', hideFromInspector: true }).toPromise(); // toPromise until we change getTagValues, getTagKeys to Observable
+    // If URL includes endpoint that supports POST method, try to use configures method
```

Comment:
```suggestion
    // If URL includes endpoint that might support POST and GET method, try to use configured method. This might fail as POST is supported only in v2.10+ .
```